addons/stock/static/src/fields/stock_move_line_x2_many_field.js

```diff
@@ -41,14 +40,10 @@ export class SMLX2ManyField extends X2ManyField {
         };
         const productName = this.props.record.data.product_id[1];
         const title = _t("Add line: %s", productName);
-        const alreadySelected = this.props.record.data.move_line_ids.records.filter((line) => line.data.quant_id?.[0]);
         const domain = [
             ["product_id", "=", this.props.record.data.product_id[0]],
             ["location_id", "child_of", this.props.context.default_location_id],
         ];
-        if (alreadySelected.length) {
-            domain.push(["id", "not in", alreadySelected.map((line) => line.data.quant_id[0])]);
-        }
```

Comment:
Is it worth trying to remove only the fully-used quants in current move lines ?
(should be useful for lot/serial products)

Reply:
I pushed a commit that implements this, but unless I handled it poorly it looks like it would be a big change so I'm not sure if we want to go forward with it.

I guess if yes, we'd also need a tour and to use proper float comparison but I wanted a first pass before doing that.

Note: in the quant selector the quantity available won't be updated based on pending changes, but I don't think that can (reasonably) be helped...
I'm also not excluding 0 availability quants, because I want them to show up when we delete a line for the relevant quant that was previously saved